subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/TasksFromProjectDependencies.java

```diff
@@ -54,4 +58,14 @@ void resolveProjectDependencies(TaskDependencyResolveContext context, Set<Projec
     public String getTaskName() {
         return taskName;
     }
+
+    @Override
+    public Set<? extends Task> getDependencies(@Nullable Task task) {
+        return taskDependencyDelegate.getDependencies(task);
```

Comment:
Nice catch! 👍

I initially thought that it should not because it is in a method that is itself marked as `@ForExternalUse` in the supertype. Thanks to your comment, I got suspicious and found out that ArchUnit does not by default consider overridden methods as annotated with the annotations that their declaration in the supertype has. So the reason why this usage was not reported is that the method reference statically resolves to `TaskDependencyInternal.getDependencies`, and this one is not annotated. This is a flaw in the test. I'll fix the ArchUnit test so that it checks for usages of the overrides, too. 

And I still think that it should be OK to use `@ForExternalUse`-API in overridden methods that are `@ForExternalUse` themselves. Do you think it's reasonable?

Reply:
I see, thanks for the explanation. Yes, I think that's reasonable 👍